rust/nasl-builtin-cryptographic/src/rsa.rs

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2024 Greenbone AG
+//
+
+// SPDX-License-Identifier: GPL-2.0-or-later
+//use md5::Digest;
+use crate::get_required_named_data;
+use core::str;
+use nasl_builtin_utils::{Context, FunctionErrorKind, NaslFunction, Register};
+use rsa::pkcs8::DecodePrivateKey;
+use rsa::signature::digest::Digest;
+use rsa::{Pkcs1v15Encrypt, Pkcs1v15Sign, RsaPrivateKey, RsaPublicKey};
+use sha1::Sha1;
+use std::fs::{self, File};
+use std::io::Read;
+
+fn rsa_public_encrypt(
+    register: &Register,
+    _: &Context,
+) -> Result<nasl_syntax::NaslValue, FunctionErrorKind> {
+    let data = get_required_named_data(register, "data")?;
+    let n = get_required_named_data(register, "n")?;
+    let e = get_required_named_data(register, "e")?;
+    let mut rng = rand::thread_rng();
+    let pub_key = RsaPublicKey::new(
+        rsa::BigUint::from_bytes_be(n),
+        rsa::BigUint::from_bytes_be(e),
+    )
+    .unwrap();
+    let enc_data = pub_key.encrypt(&mut rng, Pkcs1v15Encrypt, data).unwrap();
+    Ok(enc_data.to_vec().into())
+}
+
+fn rsa_private_decrypt(
+    register: &Register,
+    _: &Context,
+) -> Result<nasl_syntax::NaslValue, FunctionErrorKind> {
+    let data = get_required_named_data(register, "data")?;
+    let n = get_required_named_data(register, "n")?;
+    let e = get_required_named_data(register, "e")?;
+    let d = get_required_named_data(register, "d")?;
```

Comment:
Same for the parameter `pad` as above. In case of padding is disabled, use `rsa::hazmat::rsa_decrypt()` instead